src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyleRules.cs

```diff
@@ -109,5 +109,14 @@ private static bool IsInterfaceImplementation(ISymbol symbol)
 
             return false;
         }
+
+        /// <summary>
+        /// Returns true for symbols whose name starts with an underscore and
+        /// are optionally followed by an integer, such as '_', '_1', '_2', etc.
+        /// These are treated as special discard symbol names.
+        /// </summary>
+        internal static bool IsSymbolWithSpecialDiscardName(ISymbol symbol)
```

Comment:
Can we instead make this as extension method on `ISymbol` and move it to https://github.com/dotnet/roslyn/blob/master/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs?

Reply:
After I moved, IntelliSense says that the extension is available for `Features`, but not `CodeStyle` project.

Reply:
Which project/file did you move the extension to? Please ensure the file is moved to the shared project at the location mentioned above, which is imported into both CodeStyle and Features layer.

Reply:
I moved to wrong file, with the same namespace `Microsoft.CodeAnalysis.Shared.Extensions`. It really confuses me that there are so many files named `ISymbolExtension`.

Reply:
Yeah, but Roslyn has multiple layers, each requiring its own set of extensions, so it is unavoidable.